### PR TITLE
Docker install script

### PR DESCRIPTION
**Pull request by mhbuehler**
This PR adds a dockerfile, build script, and README for installing and using nGraph in docker. This replaces PR #270. The intended user is anyone who prefers to develop in docker containers over bare metal. Working with containers is becoming more and more popular for data scientists, for example, and this will improve usability for them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +139,11 @@ Note: The version of the ngraph-tensorflow-bridge is not going to be exactly
 the same as when you build from source. This is due to delay in the source 
 release and publishing the corresponding Python wheel. 
 
+### Build and run nGraph in Docker
+
+A shell script and dockerfiles are provided in the `tools` directory for easy setup in a Docker container.
```
Comment (ashahba) [inline README.md]: make `tools` a link to that folder `tools`

File: tools/Dockerfile.ubuntu18.04.install [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ==============================================================================
+
+ARG base_image
+
+FROM $base_image as base
+ARG http_proxy
+ARG https_proxy
```
Comment (ashahba) [inline tools/Dockerfile.ubuntu18.04.install]: Remove these two lines.
The build system should use `--build-arg` to handle proxies.

File: tools/Dockerfile.ubuntu18.04.install [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ARG ngtf_build_options
+
+RUN git clone https://github.com/tensorflow/ngraph-bridge.git /ngraph-bridge
+
+WORKDIR /ngraph-bridge
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
```
Comment (ashahba) [inline tools/Dockerfile.ubuntu18.04.install]: Remove these two lines.
The build system should use `--build-arg` to handle proxies.

File: tools/Dockerfile.ubuntu18.04.install [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+RUN echo "Using build options: $ngtf_build_options" && ./build_ngtf.py $ngtf_build_options
+
+FROM ubuntu:18.04
+
+ARG http_proxy
+ARG https_proxy
```
Comment (ashahba) [inline tools/Dockerfile.ubuntu18.04.install]: Remove these two lines.
The build system should use `--build-arg` to handle proxies.

File: tools/Dockerfile.ubuntu18.04.install [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
```
Comment (ashahba) [inline tools/Dockerfile.ubuntu18.04.install]: Remove these two lines.
The build system should use `--build-arg` to handle proxies.

File: tools/Dockerfile.ubuntu18.04.install [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+ENV PATH="/ngraph-bridge/build_cmake/venv-tf-py3/bin:$PATH"
+
+RUN apt-get update &&  apt-get install -y \
+    vim \
+    python3-distutils \
+    python3-pip \
+    git \
+    unzip zip wget \
+    sudo
```
Comment (ashahba) [inline tools/Dockerfile.ubuntu18.04.install]: Why `sudo` is needed? We are running dockers as root.

File: tools/Dockerfile.ubuntu18.04.install [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ENV https_proxy=$https_proxy
+
+ENV PATH="/ngraph-bridge/build_cmake/venv-tf-py3/bin:$PATH"
+
+RUN apt-get update &&  apt-get install -y \
+    vim \
+    python3-distutils \
+    python3-pip \
+    git \
+    unzip zip wget \
```
Comment (ashahba) [inline tools/Dockerfile.ubuntu18.04.install]: split these all into one per line, also sort the packages by name.
We also use these switches when possible:
```
apt-get install -y --no-install-recommends --fix-missing
```

**Comment by mhbuehler** [comment]
> Thanks @mhbuehler for the update.
> One last thing, copyright dates need to read 2020 instead now.

Thanks @ashahba - fixed.

**Pull request merged**